napari/layers/base/base.py

```diff
@@ -226,18 +226,18 @@ class Layer(KeymapProvider, MousemapProvider, ABC):
     * `_basename()`: base/default name of the layer
     """
 
-    _modeclass = Mode
+    _modeclass: Type[BaseMode] = Mode
```

Comment:
I get the idea behind this and I think this would make sense if we weren't using Enums. Ideally, we'd use something that could actually be subclassed, in which case it would make sense (e.g. pan/zoom and transform modes would be inherited). This is an issue to do with using Enums like this and should be discussed elsewhere, so don't worry about it.

What I would do instead since you aren't able to get much value of inheriting from a dummy class would be to simply use `StringEnum` instead
```suggestion
    _modeclass: Type[StringEnum] = Mode
```